### ISSUE TITLE
kubemark-500 flaking: 1 containers failed which is more than allowed 0

### ISSUE THREAD

**Issue by yujuhong** [kubernetes/kubernetes#42709]
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-kubemark-500-gce/3105

After #42264 was merged (enabling CRI in hollow nodes), the test started flaking.

Failure signature:
```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/density.go:743
Expected error:
    <*errors.errorString | 0xc423478230>: {
        s: "1 containers failed which is more than allowed 0",
    }
    1 containers failed which is more than allowed 0
not to have occurred
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/density.go:223
```

I couldn't find anything useful in the log -- what container failed or why it failed.
I think dumping more information in the test may help, since we don't have hollow node logs for kubemark-500. @wojtek-t @gmarek @shyamjvs is there any better way debug this? kubemark-100 is fine.... 

As for deflaking the build,  only [this commit](https://github.com/kubernetes/kubernetes/pull/42264/commits/c89e9cdbf109c569bff06e5db8354402bd440ffd) needs to be reverted. The rest is just fakedockerclient cleanup and they are safe to keep.

**Comment by Random-Liu** [comment]
The error is exactly what I saw when that PR was initially merged.

**Comment by liggitt** [comment]
can you go ahead and open the appropriate revert?

**Comment by yujuhong** [comment]
@liggitt I was debugging this by sshing into the kubemark-500 nodes, but as luck would have it, the most recent run passed. This successful run unblocked the submit queue. 

I will upload a PR, but I would ask that maybe we consider do one of the following:
 1. If the test doesn't flake frequently enough that it'd affect submit queue, we don't revert the PR so that we can properly debug it.
 2. Move kubemark-500 out of submit queue blocking for a day or two for proper debugging. The risk of a PR getting in and breaking kubemark 500 is pretty low (IMO) during code freeze.

**Comment by liggitt** [comment]
> The risk of a PR getting in and breaking kubemark 500 is pretty low (IMO) during code freeze.

You clearly haven't seen the size of some of the PRs going in

**Comment by yujuhong** [comment]
> You clearly haven't seen the size of some of the PRs going in

We still have kubemark-100, and kubemark-500 will still be running (just not submit queue blocking). Besides, there are only two PRs in the submit queue. Maybe I am too optimistic, but I don't think this is a ridiculous request.

**Comment by liggitt** [comment]
At least get the appropriate revert open so we can get it green if we need it

**Comment by gmarek** [comment]
I'm all with @yujuhong on this. This failure may indicate some rare race in CRI (or in it's fake implementation, but it's hard to tell at this point), so we should debug it. I'm going to enable gathering logs from kubemark-500 runs.

**Comment by wojtek-t** [comment]
+1 for having an open revert (for now, I'm fine with having it hang with "do-not-merge" label)

**Comment by wojtek-t** [comment]
The revert is here - just in case:
https://github.com/kubernetes/kubernetes/pull/42712

**Comment by wojtek-t** [comment]
>> I couldn't find anything useful in the log -- what container failed or why it failed.

> I had exactly the same question when I tried to debug it. And I could not reproduce the failure in my own environment even with 6 nodes 500 hollow nodes.

@Random-Liu @yujuhong - we just merged the PR in test-infra that should start gathering logs from hollow kubelets.

@gmarek @shyamjvs  - can you ensure in 2-3 hours that those logs are really there?

**Comment by wojtek-t** [comment]
BTW - this has also failed recently in kubemark-100:
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-kubemark-100-gce/1688
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-kubemark-100-gce/1681

**Comment by gmarek** [comment]
I think we can now revert this PR now to unblock the queue.

**Comment by yujuhong** [comment]
@wojtek-t @gmarek thanks for the help! I am closing this issue since the PR has already been reverted. Will open a new issue with more information to debug.

**Comment by wojtek-t** [comment]
@yujuhong - thanks; please cc me to the new issue and let me know if I/we can help more with debugging

**Comment by gmarek** [comment]
@wojtek-t she already did: #42736 :)
